src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -364,26 +359,55 @@ - (void)centralManager:(CBCentralManager *)central
         }
     }
 
-    if (!serviceData || [serviceData length] != 8) {
+    if (!serviceData) {
```

Comment:
Won't we get here with `serviceData == nil` even if there was no matching UUID?

In other words, shouldn't we keep track of whether we took the break above or ran out of loop?